diff --git a/common/action_code.h b/common/action_code.h
index bc40e2c..2d4673c 100644
--- a/common/action_code.h
+++ b/common/action_code.h
@@ -302,8 +302,20 @@ enum backlight_opt {
 #define ACTION_BACKLIGHT_TOGGLE()       ACTION(ACT_BACKLIGHT, BACKLIGHT_TOGGLE << 8)
 #define ACTION_BACKLIGHT_STEP()         ACTION(ACT_BACKLIGHT, BACKLIGHT_STEP << 8)
 #define ACTION_BACKLIGHT_LEVEL(level)   ACTION(ACT_BACKLIGHT, BACKLIGHT_LEVEL << 8 | level)
+
 /* Command */
-#define ACTION_COMMAND(id, opt)         ACTION(ACT_COMMAND,  (opt)<<8 | (addr))
+enum commands {
+    CMD_RECORDER_TOGGLE = 0,
+    CMD_RECORDER_START  = 1,
+    CMD_RECORDER_STOP   = 2,
+    CMD_RECORDER_PLAY   = 3,
+    CMD_RECORDER_CLEAR  = 4,
+};
+#define ACTION_COMMAND(id, opt)         ACTION(ACT_COMMAND,  (opt)<<8 | (id))
+#define ACTION_RECORD_TOGGLE(slot)      ACTION_COMMAND(CMD_RECORDER_TOGGLE, slot)
+#define ACTION_RECORD_PLAY(slot)        ACTION_COMMAND(CMD_RECORDER_PLAY, slot)
+#define ACTION_RECORD_CLEAR(slot)       ACTION_COMMAND(CMD_RECORDER_CLEAR, slot)
+
 /* Function */
 enum function_opts {
     FUNC_TAP = 0x8,     /* indciates function is tappable */
