--- keymap.c-orig	2015-02-18 22:50:30.877558467 +0100
+++ keymap.c	2015-02-18 22:52:42.553561605 +0100
@@ -85,6 +85,8 @@
 #include "keymap_micro.h"
 #elif defined(KEYMAP_CUB)
 #include "keymap_cub.h"
+#elif defined(KEYMAP_OLEMARTINORG)
+#include "keymap_olemartinorg.h"
 #else
 
 static const uint8_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
@@ -218,7 +220,7 @@
     }
 }
 
-#if defined(KEYMAP_CUB)
+#if defined(KEYMAP_CUB) || defined(KEYMAP_OLEMARTINORG)
 
 // function keymap_fn_to_action will be defined in keymap_cub.h
 
