--- matrix.c-orig	2015-02-18 22:53:54.949563330 +0100
+++ matrix.c	2015-02-18 22:56:09.685566540 +0100
@@ -166,6 +166,25 @@
     mcp23018_status = ergodox_left_leds_update();
 #endif
 
+#ifdef KEYMAP_OLEMARTINORG
+    uint8_t layer = biton32(layer_state);
+
+    ergodox_left_led_1_off();
+    ergodox_left_led_2_off();
+    switch (layer) {
+        case 1:
+            ergodox_left_led_1_on();
+            break;
+        case 2:
+            ergodox_left_led_2_on();
+            break;
+        default:
+            break;
+    }
+
+    mcp23018_status = ergodox_left_leds_update();
+#endif
+
     for (uint8_t i = 0; i < MATRIX_ROWS; i++) {
         select_row(i);
         matrix_row_t cols = read_cols(i);
