--- matrix.c-orig	2015-02-19 20:45:39.668481539 +0100
+++ matrix.c	2015-02-19 20:45:43.512481630 +0100
@@ -166,6 +166,31 @@
     mcp23018_status = ergodox_left_leds_update();
 #endif
 
+#ifdef KEYMAP_OLEMARTINORG
+    uint8_t layer = biton32(layer_state);
+
+    ergodox_board_led_off();
+    ergodox_right_led_1_off();
+    ergodox_right_led_2_off();
+    ergodox_right_led_3_off();
+    switch (layer) {
+        case 1:
+            ergodox_board_led_on();
+            break;
+        case 2:
+            ergodox_right_led_1_on();
+            break;
+        case 3:
+            ergodox_right_led_2_on();
+            break;
+        case 4:
+            ergodox_right_led_3_on();
+            break;
+        default:
+            break;
+    }
+#endif
+
     for (uint8_t i = 0; i < MATRIX_ROWS; i++) {
         select_row(i);
         matrix_row_t cols = read_cols(i);
